server/src/main/java/io/crate/execution/ddl/tables/TransportCloseTable.java

```diff
@@ -142,35 +146,44 @@ protected void masterOperation(CloseTableRequest request,
         clusterService.submitStateUpdateTask("add-block-close-table", new AddCloseBlocksTask(listener, request));
     }
 
+    public static String[] getIndices(List<RelationName> tables, @Nullable String partition) {
+        return partition != null ? new String[] {partition} : tables.stream()
+            .map(t -> t.indexNameOrAlias())
+            .toArray(String[]::new);
```

Comment:
I'd suggest to return `Indices[]` and use `indexNameExpressionResolver.concreteIndices()` here inside to avoid creating a string array for the partition.

Reply:
this method used in 2 other blocks expecting varargs.
```
@Override
protected ClusterBlockException checkBlock(CloseTableRequest request, ClusterState state) {
    return state.blocks().indicesBlockedException(ClusterBlockLevel.METADATA_WRITE,
        indexNameExpressionResolver.concreteIndexNames(state, STRICT_INDICES_OPTIONS, getIndices(request.tables(), request.partition())));
}
```
if I make it return `Index[]` I would need to almost copy that code in those 2 places.

Reply:
ok right, varargs will convert it anyhow